src/WebJobs.Extensions.DocumentDB/Bindings/DocumentDBEnumerableValueBinder.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.Documents.Client;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DocumentDB
+{
+    internal class DocumentDBEnumerableValueBinder<T> : IValueBinder
+    {
+        private DocumentDBContext _context;
+
+        public DocumentDBEnumerableValueBinder(DocumentDBContext context)
+        {
+            _context = context;
+        }
+
+        public Type Type
+        {
+            get
+            {
+                return typeof(IEnumerable<T>);
+            }
+        }
+
+        public object GetValue()
+        {
+            Uri collectionUri = UriFactory.CreateDocumentCollectionUri(_context.ResolvedAttribute.DatabaseName, _context.ResolvedAttribute.CollectionName);
+
+            List<Document> finalResults = new List<Document>();
+
+            string continuation = null;
+
+            do
+            {
+                DocumentQueryResponse response = DocumentDBUtility.RetryAsync(
+                    () => _context.Service.ExecuteNextAsync(collectionUri, _context.ResolvedAttribute.SqlQuery, continuation),
+                    _context.MaxThrottleRetries, codesToIgnore: HttpStatusCode.NotFound).Result;
+
+                if (response == null)
+                {
+                    return null;
+                }
+
+                finalResults.AddRange(response.Results);
+                continuation = response.ResponseContinuation;
+            }
+            while (!string.IsNullOrEmpty(continuation));
```

Comment:
If you want to restrict it, you'd need to put a `SELECT TOP X` in your query. So yeah it's possible to pull a billion rows.

If someone wanted to page through a massive collection, they'd have to write that themselves with DocumentClient directly. We could probably come up with a solution (for a future release) that simplifies this -- we'd probably use the pattern for Tables as well.